rekuho/src/lib/LinesView.svelte

```diff
@@ -0,0 +1,63 @@
+<!--
+    LinesView contains lines of text
+-->
+<script lang="ts" context="module">
+  import type { Writable } from "svelte/store"
+  import { writable } from "svelte/store"
+
+  import type { Vector } from "./LinearAlgebra"
+
+  export const lines: Writable<string[]> = writable([]) // contains all cached lines
+
+  export const isAlpha = (code: number): boolean =>
+    (code > 47 && code < 58) || (code > 64 && code < 91) || (code > 96 && code < 123)
+
+  const splice = (self: string, offset: number, text: string, removeCount: number = 0): string => {
+    let calculatedOffset = offset < 0 ? self.length + offset : offset
+    return (
+      self.substring(0, calculatedOffset) + text + self.substring(calculatedOffset + removeCount)
+    )
+  }
+
+  const modifyNth = <T>(list: T[], i: number, f: (_: T) => T): T[] => {
+    list[i] = f(list[i])
+    return list
+  }
```

Comment:
Weird

Reply:
Kind of functional, kind of fun. @elkowar mentioned: "probably good for a `Utils.ts` file". Don't entirely enjoy undescript miscellaneous files with very generic tools in it, but can't be helped when there's no good way to extend the standard library.